backend/app/rest/residents_routes.py

```diff
@@ -41,6 +51,17 @@ def update_resident(resident_id):
             jsonify({"date_left_error": "date_left cannot be less than date_joined"}),
             400,
         )
+    
+    # Check for the existence of a resident prior to adding them
+    fmt_resident_id = updated_resident.get("initial") + str(updated_resident.get("room_num"))
+    try:
+        res = residents_service.get_residents(False, 1, 10, fmt_resident_id)
+        if len(res["residents"]) == 1 and res["residents"][0]["id"] != resident_id:
```

Comment:
When editing a resident, there are two cases for which the result of **get_residents** could be `>=1` in size:

1. There was no change to the resident ID (i.e. a resident with ID AA123 has their building num changed ; ID remains AA123 and so when we get residents with the ID `fmt_resident_id = AA123`, then we get 1)

2. We changed another resident to have a pre-existing resident ID (there are residents AA123 and AA321 ; resident AA123 has their room number changed to 321, resulting in two duplicate entries ; thus when we fetch the residents with the ID AA321, we end up with a result of 1)

We want case #2 to be marked as INVALID but case #1 to be VALID. As such, our checking logic must check for:
- If there's already a result being returned
- If the result has the same database ID as us (if it's the same, then we're editing the same record, meaning this change is valid ; if the database IDs are **different**, then this means that there is a _different_ resident with the same formatted ID as us)